internal/lsp/server.go

```diff
@@ -1313,24 +1315,46 @@ func (l *LanguageServer) handleTextDocumentFormatting(
 		return nil, fmt.Errorf("failed to get file contents for uri %q", params.TextDocument.URI)
 	}
 
-	f := &fixes.Fmt{OPAFmtOpts: format.Opts{}}
+	// set up an in-memory file provider to pass to the fixer for this one file
+	memfp := fileprovider.NewInMemoryFileProvider(map[string][]byte{
+		params.TextDocument.URI: []byte(oldContent),
+	})
 
-	fixResults, err := f.Fix(&fixes.FixCandidate{
-		Filename: filepath.Base(uri.ToPath(l.clientIdentifier, params.TextDocument.URI)),
-		Contents: []byte(oldContent),
-	}, nil)
+	input, err := memfp.ToInput()
 	if err != nil {
-		l.logError(fmt.Errorf("failed to format file: %w", err))
+		return nil, fmt.Errorf("failed to create fixer input: %w", err)
+	}
 
-		// return "null" as per the spec
-		return nil, nil
+	li := linter.NewLinter().
+		WithUserConfig(*l.loadedConfig).
```

Comment:
This is a nil-dereference panic if no config is provided. I know because I did the same mistake on my branch 😆